### PR TITLE
Revert "Enable non-blittable struct returns on UnmanagedCallersOnly (#45625)"

### PR DESCRIPTION
**Pull request by safern**
This commit caused the rolling build to break when running tests. There are various symptoms here like crashes or test failures because a wrong value is returned.

```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
         at System.Buffer.Memmove(Byte ByRef, Byte ByRef, UIntPtr)
         at System.Runtime.InteropServices.Marshal.CopyToManaged[[System.Byte, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](IntPtr, Byte[], Int32, Int32)
         at System.Reflection.BlobUtilities.ReadBytes(Byte*, Int32)
```

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-db730bf754b04385ad/PayloadGroup0/console.22213a03.log?sv=2019-07-07&se=2021-01-05T20:49:15Z&sr=c&sp=rl&sig=Ms13Vjh0e7qy3habeZFOis96Fx%2BgijRAnLgpkflACvk%3D

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-9dcde7f053014a80a1/System.Collections.Tests/console.82eb4f6b.log?sv=2019-07-07&se=2021-01-05T19%3A40%3A21Z&sr=c&sp=rl&sig=VAqIZP6UszpjxQrfkUsXjn%2BZXls0jLc8ly3k63YCR6c%3D

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-0e44ba7fc5b24350b4/System.Runtime.Numerics.Tests/console.f91e83a2.log?sv=2019-07-07&se=2021-01-05T19%3A38%3A53Z&sr=c&sp=rl&sig=9lF%2FPpk6%2FpUZ2bAEAcQtkoZTkNFuKVINxeg%2BrH8ZtRI%3D

The PR Build for #45625 was green. The reason for that is that this only manifests when running on 64 bit on release and it wasn't caught due to not having any configuration for libraries on release running tests on this archs. @jkoritzinsky will follow up on that to run Libraries Release on a Checked CoreCLR on PRs.

cc: @jkoritzinsky @jkotas @akoeplinger

### CODE REVIEW

File: src/coreclr/inc/jiteeversionguid.h [hunk 1]
```diff
@@ -31,11 +31,11 @@
 //
 //////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-constexpr GUID JITEEVersionIdentifier = { /* a7bb194e-4e7c-4850-af12-ea9f30ea5a13 */
-    0xa7bb194e,
-    0x4e7c,
-    0x4850,
-    {0xaf, 0x12, 0xea, 0x9f, 0x30, 0xea, 0x5a, 0x13}
+constexpr GUID JITEEVersionIdentifier = { /* 790de1e5-1426-4ecf-939c-2cd60445c219 */
```
Comment (safern) [inline src/coreclr/inc/jiteeversionguid.h:34]: @dotnet/jit-contrib what should I do here, should I revert it to the old value before this change?

**Comment by safern** [comment]
I confirmed that this broke the build by queueing a build without that commit and everything passed:
https://dnceng.visualstudio.com/public/_build/results?buildId=924035&view=results

**Review by jkotas — APPROVED** [review]

File: src/coreclr/inc/jiteeversionguid.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/inc/jiteeversionguid.h:34]: The safest thing to do is to generate a new guid as the comment above describes.

Comment (safern) [reply src/coreclr/inc/jiteeversionguid.h:34]: Sounds good. I'll do that.

**Comment by BruceForstall** [comment]
@kunalspathak maybe this is related to the problem of the superpmi collection being broken?

**Comment by jkotas** [comment]
Fixed by #46176

**Pull request closed without merging**